### PR TITLE
@ashkan18 => [WIP] Add message details (as Relay connection)

### PR DESCRIPTION
**Pull request by mzikherman**
Depends on https://github.com/artsy/impulse/pull/262/files , so this was basically coded blind.

But...something like this. For instance, replacing `messages` with `message_details` in Emission (cc @sarahscott ), will enable the text of each message to be rendered (using `raw_text` in place of the `snippet` stuff). We can also then show the attachments, etc.

@sarahscott we now have `MessageType` and `MessageDetailType` (what I named the Radiation stuff here). I think we can basically just use `MessageDetail` since that includes the full text (instead of just the snippet helper that we were relying on currently). We should sync up about that b/c we'll need to make some changes in Emission probably.

### CODE REVIEW

**Comment by mzikherman** [comment]
(Once https://github.com/artsy/impulse/pull/262/files is merged I'll finish this up for realsies).

**Comment by sarahscott** [comment]
This looks awesome! What if we just replace the existing `MessageType` completely and make `MessageDetailType`s the new `messages`? Emission is the only app that uses any of this at the moment and we could update its schema to correspond with this change.

**Comment by sarahscott** [comment]
I can get a pr in emission today so that a change to the message schema wouldn't be breaking. Also, it seems that semaphore isn't running?

File: schema/me/conversation.js [hunk 1]
```diff
@@ -59,6 +59,48 @@ export const MessageType = new GraphQLObjectType({
...
+  fields: {
+    id: {
+      description: "Radiation id.",
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    attachments: {
+      description: "Attachments.",
+      type: new GraphQLList(AttachmentType),
+    },
+    raw_text: {
```
Comment (ashkan18) [inline schema/me/conversation.js]: should we add `metadata` too? im sure it'll be useful down the line.

Comment (sarahscott) [inline schema/me/conversation.js]: Would `metadata` include a timestamp? We definitely need that.

**Comment by mzikherman** [comment]
> This looks awesome! What if we just replace the existing MessageType completely and make MessageDetailTypes the new messages? Emission is the only app that uses any of this at the moment and we could update its schema to correspond with this change.

@sarahscott yup this is what I was thinking since we're going to be just fetching all the messages/details anyway. In that case if you get that Emission PR ready I can get this ready (and get that Impulse one merged) and we can sync them, but no huge rush. I was getting more comfortable in Emission the last day or two and I could probably make those changes too.

File: schema/me/conversation.js (hunk 1 above)
Comment (ashkan18) [inline schema/me/conversation.js]: like `created_at` timestamps? it should be part of `message` itself, we can add those.

Metadata basically has some information about the message itself, basically tells us if this message was an invoice message or an initial inquiry or etc..

Comment (ashkan18) [inline schema/me/conversation.js]: @mzikherman i would also add `to`, `from` and `cc`

Comment (sarahscott) [inline schema/me/conversation.js]: Yeah a `created_at` timestamp should be fine; here's what the emission spec looks like (maybe we can format the timestamp here in mp as well?) :
screen shot 2017-06-21 at 3 53 34 pm

**Comment by mzikherman** [comment]
Closing for now...

**Pull request closed without merging**